docs/usage.md

```diff
@@ -0,0 +1,107 @@
+# Usage
+
+There are two main ways to use nion from React: as a _hook_, or as a _HOC_ (higher-order component). The HOC usage itself comes in two forms: modern HOC and legacy decorators.
+
+## As a hook
+
+When writing new code, the recommended way to use nion is the `useNion` [hook](https://reactjs.org/docs/hooks-intro.html). Note that the `dataKey` in the declaration is required.
+
+```js
+import { useNion } from '@nion/nion'
+
+export const UserContainer = () => {
+    const [currentUser, actions, request] = useNion({
+        dataKey: 'currentUser',
+        endpoint: 'https://patreon.com/api/current_user',
+    })
+
+    const loadButton = <Button onClick={() => actions.get()}>Load</Button>
+
+    return (
+        <Card>
+            {request.isLoading ? <LoadingSpinner /> : loadButton}
+            {currentUser && <UserCard user={currentUser} />}
+        </Card>
+    )
+}
+```
+
+## As a modern HOC
+
+Since a class component does not support hooks, when working with one, use the `nion` [higher-order component (HOC)](https://reactjs.org/docs/higher-order-components.html) instead. Note that the `dataKey` can be omitted from the declaration, since it is inferred from the property name that contains the declaration.
+
+```js
+import { nion } from '@nion/nion'
+
+class UserContainer extends Component {
+    render() {
+        const { currentUser } = this.props.nion
+        const { request, actions, data } = currentUser
+
+        const loadButton = <Button onClick={() => actions.get()}>Load</Button>
+
+        return (
+            <Card>
+                {request.isLoading ? <LoadingSpinner /> : loadButton}
+                {data && <UserCard user={data} />}
+            </Card>
+        )
+    }
+}
+
+export default nion({
```

Comment:
Similarly want to know if we can just do `export default nion('currentUser')` if for `ChildComponent` in the tree

Reply:
Will do more research to find out.

Reply:
Yes you can.

Reply:
(Just want this mentioned in the docs too since this is a very common use case)